src/ServiceControl.Monitoring/Metrics/Raw/RawMessageSerializer.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace ServiceControl.Monitoring.Metrics.Raw
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+
+    public class RawMessageSerializer<T> where T : RawMessage, new()
+    {
+        static readonly object[] NoMessages = new object[0];
+
+        public void Serialize(object message, Stream stream)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected object[] DeserializeRawMessage(Stream stream, Func<long, BinaryReader, T, bool> store)
```

Comment:
`store` could be an `abstract` method of this type (`RawMessageSerializer` is used only as a base class) removing the need of passing the storing delegate.